lib/jobs/local-catalog.js

```diff
@@ -0,0 +1,175 @@
+     * @param {String} taskId
+     * @constructor
+     */
+    function LocalCatalogJob(options, context, taskId) {
+        LocalCatalogJob.super_.call(this, 
+                                   logger,
+                                   options, 
+                                   context, 
+                                   taskId);
+
+        assert.string(this.context.target);
+        assert.arrayOfString(this.options.commands);
+
+        this.nodeId = this.context.target;
+        this.commands = _.isArray(options.commands) ? options.commands : [options.commands];
+        this.acceptedResponseCodes = options.acceptedResponseCodes[0] || 1;
+    }
+    util.inherits(LocalCatalogJob, BaseJob);
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype._run = function() {
+        var self = this;
+        var cmdResp = [];
+
+        waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            assert.ok(node, 'No node for local catalog');
+            
+            return Promise.each(self.commands, function(cmd) {
+                return self.runCommand(cmd)
+                .then(function(ret) {
+                    cmdResp.push(ret);
+                });
+            });
+        })
+        .then(function() {
+            return cmdResp;
+        })
+        .then(self.handleResponse.bind(self))
+        .then(function() {
+            self._done();
+        })
+        .catch(function(e) {
+            self._done(e);
+        });
+            
+    };
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype.runCommand = function (cmd) {
+        var self = this;
+        var args = cmd.split(' ');
+        var program = args[0];
+        if(program === "sudo") {
+            program = args[1];
+            args.shift();
+        }
+        args.shift();
+        return Promise.resolve()
+        .then(function() {
+            var childProcess = new ChildProcess(
+                                            program,
+                                            args,
+                                            {},
+                                            self.acceptedResponseCodes);
+
+            logger.debug("Sending command to node.", {
+                command: cmd,
+                nodeId: self.nodeId
+            });
+
+            return childProcess.run({ retries: 0, delay: 0 });
+        })
+        .then(function(ret) {
+            ret.cmd = cmd;
+            if (ret.stderr) {
+                ret.error = self.acceptedResponseCodes;
+            }
+            logger.debug("Received respond from node.", {
+                nodeId: self.nodeId,
+                error: ret.error
+            });
+            return ret;
+        });
+    };
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype.handleResponse = function(result) {
+        var self = this;
+
+        return parser.parseTasks(result)
```

Comment:
We also support regexes to match against commands, in cases where we don't want to/can't be explicit (like ipmi user N).